frontend/src/components/tabs/InputOutputTab.tsx

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2021 The Kubeflow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Artifact,
+  ArtifactCustomProperties,
+  ArtifactProperties,
+  Execution,
+  getMetadataValue,
+  getResourceProperty,
+} from '@kubeflow/frontend';
+import { Struct } from 'google-protobuf/google/protobuf/struct_pb';
+import React from 'react';
+import { useQuery } from 'react-query';
+import { ErrorBoundary } from 'src/atoms/ErrorBoundary';
+import { color, commonCss, padding, spacing } from 'src/Css';
+import { getInputArtifactsInExecution, getOutputArtifactsInExecution } from 'src/lib/MlmdUtils';
+import { KeyValue } from 'src/lib/StaticGraphParser';
+import { stylesheet } from 'typestyle';
+import { ArtifactLink } from '../ArtifactLink';
+import Banner from '../Banner';
+import DetailsTable, { ValueComponentProps } from '../DetailsTable';
+import { ExecutionTitle } from './ExecutionTitle';
+
+const css = stylesheet({
+  key: {
+    color: color.strong,
+    flex: '0 0 50%',
+    fontWeight: 'bold',
+    maxWidth: 300,
+  },
+  row: {
+    borderBottom: `1px solid ${color.divider}`,
+    display: 'flex',
+    padding: `${spacing.units(-5)}px ${spacing.units(-6)}px`,
+  },
+  valueText: {
+    flexGrow: 1,
+    overflow: 'hidden',
+    overflowWrap: 'break-word',
+  },
+});
+
+type ParamList = Array<KeyValue<string>>;
+
+export interface IOTabProps {
+  execution: Execution;
+}
+
+export function InputOutputTab({ execution }: IOTabProps) {
+  const executionId = execution.getId();
+
+  // Retrieves input and output artifacts from Metadata store.
+  const {
+    isSuccess: isOutputArtifactSuccess,
+    error: outputArtifactError,
+    data: outputArtifactData,
+  } = useQuery<Artifact[], Error>(
+    ['execution_output_artifact', { id: executionId }],
+    () => getOutputArtifactsInExecution(execution),
+    { staleTime: Infinity },
+  );
+
+  const {
+    isSuccess: isInputArtifactSuccess,
+    error: inputArtifactError,
+    data: inputArtifactData,
+  } = useQuery<Artifact[], Error>(
+    ['execution_input_artifact', { id: executionId }],
```

Comment:
That is a good point of sending one request rather than two for event lists and artifact lists. I am going to merge the async logics into one query to get all artifacts, and filter them in local client.